chalk-engine/src/logic.rs

```diff
@@ -115,7 +115,14 @@ impl<C: Context> Forest<C> {
         assert!(self.stack.is_empty());
 
         match self.ensure_answer_recursively(context, table, answer) {
-            Ok(EnsureSuccess::AnswerAvailable) => Ok(()),
+            Ok(EnsureSuccess::AnswerAvailable) => {
+                let answer = self.answer(table, answer);
+                for delayed_subgoal in answer.delayed_subgoals.clone() {
```

Comment:
I haven't written tests for  variant 2 or 3 yet, but no tests hit this. So I'm wondering what conditions this could happen.